### PR TITLE
Add sriovfec pkg

### PR DESCRIPTION
**Pull request by mcornea**
Add sriovfec pkg together with sriov-fec-operator API. This is needed since vendoring doesn't work due to older version of sigs.k8s.io/controller-runtime pkg[0] than what is used in eco-goinfra.

[0] https://github.com/smart-edge-open/sriov-fec-operator/blob/main/go.mod#L27

### CODE REVIEW

**Comment by mcornea** [comment]
Hi @trewest @kononovn Could you please review this change when you get a moment and let me know if there's a better way to deal with this situation? Thanks!

File: pkg/sriov-fec/nodeconfig.go [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+)
+
+const (
+	sriovfecnodeconfig = "SriovFecNodeConfig"
+)
+
+// Builder provides struct for the SriovFecNodeConfig object containing connection to
+// the cluster and the SriovFecNodeConfig definitions.
+type Builder struct {
+	Definition *sriovfectypes.SriovFecNodeConfig
```
Comment (kononovn) [inline pkg/sriov-fec/nodeconfig.go:26]: Please add comments to public elements of the struct

File: pkg/sriov-fec/nodeconfig.go [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+	Object     *sriovfectypes.SriovFecNodeConfig
+	apiClient  *clients.Settings
+	errorMsg   string
+}
+
+// AdditionalOptions additional options for sriovfecnodeconfig object.
+type AdditionalOptions func(builder *Builder) (*Builder, error)
+
+// NewBuilder creates a new instance of Builder.
+func NewBuilder(apiClient *clients.Settings, name, nsname string, label map[string]string) *Builder {
```
Comment (kononovn) [inline pkg/sriov-fec/nodeconfig.go:36]: Please re-name NewBuilder to NewNodeConfigBuilder.

**Review by kononovn — CHANGES_REQUESTED** [review]
Please check few comments above.

File: pkg/sriov-fec/nodeconfig.go (hunk 1 above)
Comment (mcornea) [reply pkg/sriov-fec/nodeconfig.go:27]: Thanks, added comments.

File: pkg/sriov-fec/nodeconfig.go [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const (
+	sriovfecnodeconfig = "SriovFecNodeConfig"
+)
+
+// Builder provides struct for the SriovFecNodeConfig object containing connection to
+// the cluster and the SriovFecNodeConfig definitions.
+type Builder struct {
```
[the remark this answers is not in the document]
Comment (mcornea) [reply pkg/sriov-fec/nodeconfig.go]: Updated to NodeConfigBuilder.

File: pkg/sriov-fec/nodeconfig.go (hunk 2 above)
Comment (mcornea) [reply pkg/sriov-fec/nodeconfig.go]: Re-named to NewNodeConfigBuilder.

File: pkg/sriov-fec/nodeconfig.go [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+
+// AdditionalOptions additional options for sriovfecnodeconfig object.
+type AdditionalOptions func(builder *NodeConfigBuilder) (*NodeConfigBuilder, error)
+
+// NewNodeConfigBuilder creates a new instance of NodeConfigBuilder.
+func NewNodeConfigBuilder(
+	apiClient *clients.Settings,
+	name, nsname string,
+	label map[string]string,
+) *NodeConfigBuilder {
```
Comment (kononovn) [inline pkg/sriov-fec/nodeconfig.go:45]: Please format function according to code convention.
func NewNodeConfigBuilder(
	apiClient *clients.Settings,
	name, nsname string,
	label map[string]string) *NodeConfigBuilder {

File: pkg/sriov-fec/nodeconfig.go [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+		builder.errorMsg = "SriovFecNodeConfig 'nsname' cannot be empty"
+	}
+
+	return &builder
+}
+
+// Pull retrieves an existing SriovFecNodeConfig.io object from the cluster.
+func Pull(apiClient *clients.Settings, name, nsname string) (*NodeConfigBuilder, error) {
+	glog.V(100).Infof(
+		"Pulling SriovFecNodeConfig.io object name:%s in namespace: %s", name, nsname)
```
Comment (kononovn) [inline pkg/sriov-fec/nodeconfig.go:82]: Please add space `name: %s`

File: pkg/sriov-fec/nodeconfig.go (hunk 4 above)
Comment (mcornea) [reply pkg/sriov-fec/nodeconfig.go]: Thanks, updated.

File: pkg/sriov-fec/nodeconfig.go (hunk 5 above)
Comment (mcornea) [reply pkg/sriov-fec/nodeconfig.go]: Updated.

File: pkg/sriov-fec/nodeconfig.go [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+		apiClient: apiClient,
+		Definition: &sriovfectypes.SriovFecNodeConfig{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
```
Comment (kononovn) [inline pkg/sriov-fec/nodeconfig.go:93]: Small update. We've decided to update Pull function.
Can you please refactor the function to the following format:

```
if name == "" {
                return nil, fmt.Error("The name of the SriovFecNodeConfig is empty")
	}

if nsname == "" {
                return nil, fmt.Error("The namespace of the SriovFecNodeConfig is empty")
	}
```

**Review by kononovn — CHANGES_REQUESTED** [review]
Please few comment above.

File: pkg/sriov-fec/nodeconfig.go (hunk 6 above)
Comment (mcornea) [reply pkg/sriov-fec/nodeconfig.go:93]: Sure, updated.

**Review by kononovn — APPROVED** [review]
/lgtm

**Review by cdvultur — APPROVED** [review]
/lgtm

**Pull request merged**